broadlink/climate.py

```diff
@@ -238,3 +238,324 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
             input_payload.append(int(weekend[i]["temp"] * 2))
 
         self.send_request(input_payload)
+
+class tornado(device):
+    """Controls Tornado TOP SQ X series air conditioners."""
+    def __init__(self, *args, **kwargs):
+        device.__init__(self, *args, **kwargs)
+        self.type = "Tornado air conditioner"
+    
+    def _decode(self, response) -> bytes:
+        payload = self.decrypt(bytes(response[0x38:]))
+        return payload
+    
+    def _calculate_checksum(self, packet:bytes, target:int=0x20017) -> tuple:
+        """Calculate checksum of given array,
+        by adding little endian words and subtracting from target.
+        
+        Args:
+            packet (list/bytearray/bytes): 
+        """
+        result = target - (sum([v if i % 2 == 0 else v << 8 for i, v in enumerate(packet)]) & 0xffff)
+        return (result & 0xff, (result >> 8) & 0xff)
+
+    def _send_short_payload(self, payload:int) -> bytes:
+        """Send a request for info from A/C unit and returns the response.
+        0 = GET_AC_INFO, 1 = GET_STATES, 2 = GET_SLEEP_INFO, 3 = unknown function
+        """
+        header = bytearray([0x0c, 0x00, 0xbb, 0x00, 0x06, 0x80, 0x00, 0x00, 0x02, 0x00])
```

Comment:
I think [0x0c, 0x00] is the payload length.

Reply:
The first two bytes carry the payload length. Three samples you gave me:

a) 0c 00 **bb 00 06 80 00 00 02 00 11 01 2b 7e** 00 00
b) 22 00 **bb 00 07 00 00 00 18 00 01 21 c0 24 00 00 00 3c 00 3e 00 00 00 00 00 00 00 00 00 00 00 00 00 07 64 38** 00 00 00 00 00 00 00 00 00 00 00 00
c) 19 00 **bb 00 07 00 00 00 0f 00 01 11 9f e4 2d 60 00 20 00 00 00 00 00 00 00 60 89** 00 00 00 00 00

We can use:
```
# For reading
payload_size = int.from_bytes(packet[:2], "little")

# For sending
payload_size = int.to_bytes(len(payload), 2, "little")
```
It reminds me that I have to rework the rm4 class because this is what happened there. In the new firmware the payload is prefixed with its length.